### PR TITLE
add self contained kubeconfig flag

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/roadmap/issues/542

When creating a pull request for `kubectl-gs`, please consider:

- Provide a link to the issue, and please describe the goal you are trying to accomplish in this description.
- SIG UX cares about almost all changes here and is always happy to offer feedback. Simply request a review.
- Add a user-friendly description of your change to `CHANGELOG.md`.
- Our public [kubectl-gs documentation](https://docs.giantswarm.io/ui-api/kubectl-gs/) may need an update to reflect the change you are making.

### CODE REVIEW

**Comment by marians** [comment]
Looks good!

One nitpick I have is the unnecessary empty `certificate-authority` attribute.

image

I guess that's a problem in our kubeconfig library. I think `omitempty` is missing here: https://github.com/giantswarm/kubeconfig/blob/master/types.go#L42-L43

**Comment by marians** [comment]
I just noticed that we are not checking whether an existing output file gets overwritten. Let's please do that early, and show an error if the file already exist.

File: cmd/login/clientcert.go [hunk 1]
```diff
@@ -11,8 +12,10 @@ import (
 	corev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/apiextensions/v3/pkg/label"
 	"github.com/giantswarm/backoff"
+	kubeconfig "github.com/giantswarm/kubeconfig/v3"
```
Comment (axbarsan) [inline cmd/login/clientcert.go:15]: I don't think we need this.
`"k8s.io/client-go/tools/clientcmd/api"` provides everything we need and we already use it in multiple places (even in this file)

File: cmd/login/runner.go [hunk 2]
```diff
@@ -415,7 +415,7 @@ func (r *runner) createClusterClientCert(ctx context.Context) error {
 	}
 
 	// Store client certificate credential into the kubeconfig.
-	contextName, contextExists, err := storeCredential(r.k8sConfigAccess, r.fs, clientCertResource, secret, clusterBasePath)
+	contextName, contextExists, err := storeCredential(r.k8sConfigAccess, r.fs, r.flag.WCSelfContained, clientCertResource, secret, clusterBasePath)
```
Comment (axbarsan) [inline cmd/login/runner.go:418]: Is it possible to have an if statement here where we decide whether we `store` or `print` the credential? It's a bit confusing to have printing done in the `storeCredential` function

Comment (anvddriesch) [reply cmd/login/runner.go:418]: Sure, we can do that, too! :)

File: cmd/login/clientcert.go (hunk 1 above)
Comment (anvddriesch) [reply cmd/login/clientcert.go:15]: Cool, I'll use that one instead!

File: cmd/login/clientcert.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/giantswarm/backoff"
 	"github.com/giantswarm/microerror"
 	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
```
Comment (axbarsan) [inline cmd/login/clientcert.go:16]: Sorry, I missed this before. Can we use `sigs.k8s.io/yaml`? It's mentioned in [fmt](https://github.com/giantswarm/fmt/blob/master/go/yaml_marshalling.md)

Comment (anvddriesch) [reply cmd/login/clientcert.go:16]: Yeah I also just noticed that. I copied this function from gsctl so the older dependencies came along.

File: cmd/login/clientcert.go [hunk 4]
```diff
@@ -245,16 +244,12 @@ func printCredential(k8sConfigAccess clientcmd.ConfigAccess, fs afero.Fs, filePa
...
-	if err != nil {
-		return "", false, microerror.Mask(err)
-	}
 	if exists, err := afero.Exists(fs, filePath); exists {
 		return "", false, microerror.Maskf(fileExistsError, "The destination file %s already exists. Please specify a different destination.", filePath)
 	} else if err != nil {
 		return "", false, microerror.Mask(err)
 	}
-	err = afero.WriteFile(fs, filePath, bytes, 0600)
+	err = clientcmd.WriteToFile(kubeconfig, filePath)
```
Comment (axbarsan) [inline cmd/login/clientcert.go:252]: Nice 👍

**Review by axbarsan — APPROVED** [review]

**Pull request merged**